### PR TITLE
[FEATURE] credit auth

### PR DESCRIPTION
**Pull request by sentientwaffle**
Fixes https://github.com/interledger/five-bells-ledger/issues/173

It is disabled by default, so this is not a breaking change.

### CODE REVIEW

File: src/models/transfers.js [hunk 1]
```diff
@@ -181,19 +181,23 @@ function validateIsAffectedAccount (account, transfer) {
...
-function validateAuthorizations (authorizedAccount, transfer, previousDebits) {
-  if (previousDebits && transfer.debits.length !== previousDebits.length) {
-    throw new UnprocessableEntityError('Invalid change in number of debits')
+/**
+ * @param {Account} authorizedAccount
+ * @param {Funds} funds
+ * @param {Funds|undefined} previousFunds
+ * @param {String} type "debit" or "credit"
+ */
+function validateAuthorizations (authorizedAccount, fundsList, previousFunds, fundsType) {
```
Comment (clark800) [inline src/models/transfers.js]: Another naming idea for this: we used `adjustment` as a hypernym of `debit` and `credit` in RippleAPI

**Comment by MatthewPhinney** [comment]
Code LGTM, although in general I'm kind of wary of introducing command line flags that change the behavior of the app. Maybe one flag is fine, but as we add more, the app gets much more difficult to debug. If we have even two or three of these flags, we could end up with the exact same code producing quite different results, depending on what flags are enabled. How strong is the business use case for allowing this to be enabled/disabled? Is this something that we definitely don't want in all instances of the app?

File: src/lib/config.js [hunk 2]
```diff
@@ -10,7 +10,12 @@ function isRunningTests () {
 }
 
 function loadConfig () {
-  const localConfig = {}
+  const localConfig = {
+    features: {
+      hasCreditAuth: Config.getEnv('FEATURE_CREDIT_AUTH')
```
Comment (lumberj) [inline src/lib/config.js]: `Config.castBool(Config.getEnv(envPrefix, 'FEATURE_CREDIT_AUTH'))`

File: src/models/transfers.js (hunk 1 above)
Comment (sentientwaffle) [inline src/models/transfers.js]: I'm open to that, but its out of the scope of this PR -- the name 'funds' refers to the schema in five-bells-shared.

File: .eslintrc [hunk 3]
```diff
@@ -1,2 +1,4 @@
 # We use JavaScript Standard Style (https://github.com/feross/standard)
 extends: standard
+rules:
```
Comment (clark800) [inline .eslintrc]: Unnecessary now

**Comment by clark800** [comment]
@sentientwaffle This pull request has been sitting idle for a while, can you respond to the questions?

**Comment by sentientwaffle** [comment]
@MatthewPhinney https://github.com/interledger/five-bells-ledger/issues/173 specifies an env variable feature flag, in order to prevent a breaking change.

@lumberj good catch; fixed

File: test/putTransferSpec.js [hunk 4]
```diff
@@ -1263,4 +1264,84 @@ describe('PUT /transfers/:id', function () {
       .expect(401)
       .end()
   })
+
+  describe('FEATURE_CREDIT_AUTH=true', function () {
+    // Kludge to work around immutable config
```
Comment (lumberj) [inline test/putTransferSpec.js]: This is no longer necessary with five-bells-shared@12

Comment (lumberj) [inline test/putTransferSpec.js]: (config mutable during unit testing)

**Comment by justmoon** [comment]
@MatthewPhinney Good comment. I think in this situation a feature flag is the right approach. Eventually we probably want this to be always enabled, but there are a lot of changes that need to happen to other components. So adding this behind a feature flag for now allows us to start working on those other components without any time pressure from an open PR or a broken implementation.

When everything has migrated to the new behavior we can start defaulting to having this feature enabled and eventually remove the flag altogether.

**Comment by MatthewPhinney** [comment]
Makes sense. LGTM.

**Pull request merged**